src/Widgets/Proxy/ProxyPage.vala

```diff
@@ -53,21 +77,140 @@ namespace Network.Widgets {
+            add (permission_infobar);
+#endif
+            add (control_box);
             add (stackswitcher);
             add (stack);
 
             show_all ();
 
             stack.visible_child = configuration_page;
+
+#if ENABLE_SYSTEMWIDE_PROXY
+            update_infobar_visibility ();
+#endif
+        }
+
+#if ENABLE_SYSTEMWIDE_PROXY
+        private void on_proxy_settings_changed () {
+            if (system_wide_available) {
+                if (proxy_settings.mode == "manual") {
+                    if (http_settings.host != "" && http_settings.port > 0) {
+                        system_proxy_service.set_proxy ("http", "http://%s:%d".printf (http_settings.host, http_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("http", "");
+                    }
+
+                    if (https_settings.host != "" && http_settings.port > 0) {
+                        system_proxy_service.set_proxy ("https", "https://%s:%d".printf (https_settings.host, https_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("https", "");
+                    }
+
+                    if (ftp_settings.host != "" && ftp_settings.port > 0) {
+                        system_proxy_service.set_proxy ("ftp", "ftp://%s:%d".printf (ftp_settings.host, ftp_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("ftp", "");
+                    }
+
+                    if (socks_settings.host != "" && socks_settings.port > 0) {
+                        system_proxy_service.set_proxy ("socks", "socks://%s:%d".printf (socks_settings.host, socks_settings.port));
+                    } else {
+                        system_proxy_service.set_proxy ("socks", "");
+                    }
+
+                    system_proxy_service.set_no_proxy (string.joinv (",", proxy_settings.ignore_hosts));
+                } else {
+                    system_proxy_service.set_proxy ("http", "");
+                    system_proxy_service.set_proxy ("https", "");
+                    system_proxy_service.set_proxy ("ftp", "");
+                    system_proxy_service.set_proxy ("socks", "");
+                    system_proxy_service.set_no_proxy ("");
+                }
+            }
         }
 
-        protected override void control_switch_activated () {
+        private static Polkit.Permission? get_permission () {
+            try {
+                var permission = new Polkit.Permission.sync ("org.pantheon.switchboard.networking.setproxy",
+                                                             new Polkit.UnixProcess (Posix.getpid ()));
+
+                return permission;
+            } catch (Error e) {
+                critical (e.message);
+                return null;
+            }
+        }
+
+        private void update_infobar_visibility () {
+            if (configuration_page.manual_mode) {
```

Comment:
Could we simplify this to: `permission_infobar.visible = configuration_page.manual_mode`?